query-service-api/src/main/proto/request.proto

```diff
@@ -29,6 +29,7 @@ message QueryRequest {
   int32 limit = 7;
   int32 offset = 8;
   bool distinctSelections = 9;
+  bool interactive = 10;
```

Comment:
What is your thought process here? Do we need this?

Reply:
The idea here is allow the client to specify whether the query response is expected in real-time or interactive. 
- In case of former, the query will be executed by real-time processing systems like pinot/postgres. UI will make a blocking sync call with timeout.
- In case of latter, query will be executed by interactive query processing systems like Trino. UI can potentially make an async call to consume the response.
To begin with, this will allow UI to host 2 Explorer page - the current one talking to Pinot (with interactive = false) and the other talking to Trino (with interactive = true).

Reply:
Can we keep the API interface as is for now as we're still uncertain about the final UI flow. So, we don't know what the API change may or may not requires. 

There are two options I can think for now:      
- we can put an additional 'explore page v2,' behind a feature flag. Additionally, we can pass this information as part of the QS's RequestContext to allow for dynamic selection?
- The other option would involve defining a time-based window handler. If the queries is defined for spanEventView and it has more than 3 days range, we could defined trino based handler.

Reply:
> we can put an additional 'explore page v2,' behind a feature flag.
One of the motivations behind `interactive` flag is to drive explore v2 page itself.

> The other option would involve defining a time-based window handler. If the queries is defined for spanEventView and it has more than 3 days range, we could defined trino based handler.
There could be multiple conditions involved in deciding whether query should be real-time or interactive, which we still don't know yet. It could be a combination of time-range, the kind of columns (their cardinality), the kind of functions (e.g. regex). 
My thought is that this decision around nature of query (realtime vs interactive) can't be just taken by backend alone, since UI needs to be aware of that decision so that it can show the appropriate UX. Hence it would be simpler if the inputs to make that decision are exposed in a configurable manner to UI. That means, for backend it translates to a boolean query mode.

Reply:
As discussed, removing this field for now till we get more clarity on user workflow.